ts/themes/SessionTheme.tsx

```diff
@@ -18,3 +20,39 @@ export const SessionTheme = ({ children }: { children: any }) => (
     {children}
   </>
 );
+
+export function getOppositeTheme(themeName: string) {
+  if (themeName.includes('dark')) {
+    return themeName.replace('dark', 'light');
+  }
+  if (themeName.includes('light')) {
+    return themeName.replace('light', 'dark');
+  }
+  // If neither 'dark' nor 'light' is in the theme name, return the original theme name.
+  return themeName;
+}
+
+export async function checkThemeCongruency(): Promise<boolean> {
+  const theme = window.Events.getThemeSetting();
+
+  return new Promise(resolve => {
+    ipcRenderer.send('get-native-theme');
+    ipcRenderer.once('send-native-theme', (_, shouldUseDarkColors) => {
+      const isMismatchedTheme =
+        (shouldUseDarkColors && theme.includes('light')) ||
+        (!shouldUseDarkColors && theme.includes('dark'));
+      if (isMismatchedTheme) {
```

Comment:
we should create a function to hide the light/dark check and use it rather than duplicating this code everywhere.
Maybe add a file in `ts/utils` named `theme.ts`

and add a `isDarkTheme(theme: ThemeStateType)` and `isLightTheme(theme: ThemeStateType)` there.

Then, you can update the existing selectors to use those too in `ts/state/selectors/theme.ts`
```
export const isDarkTheme = (state: StateType): boolean => isDarkTheme(state.theme); // you might need to  use another name as there would be functions with the same name
export const isLightTheme = (state: StateType): boolean => isLightTheme(state.theme);
``` 

same, move your getOppositeTheme to that file so you can make the ActionPanel call use it too 

```ts

  const handleClick = () => {
    if (type === SectionType.Profile) {
      dispatch(editProfileModal({}));
    } else if (type === SectionType.ColorMode) {
      const currentTheme = String(window.Events.getThemeSetting());
      const newTheme = (
        isDarkMode ? currentTheme.replace('dark', 'light') : currentTheme.replace('light', 'dark')
      ) as ThemeStateType;

      // We want to persist the primary color when using the color mode button
      void switchThemeTo({
        theme: newTheme,
        mainWindow: true,
        usePrimaryColor: true,
        dispatch,
      });
    } else if (type === SectionType.PathIndicator) {
      // Show Path Indicator Modal
      dispatch(onionPathModal({}));
    } else {
      // message section
      dispatch(clearSearch());
      dispatch(showLeftPaneSection(type));
      dispatch(resetOverlayMode());
    }
  };
```

Reply:
Added a isThemeMismatched() function to theme.ts refactored usage of this repeated code